### PR TITLE
Zombie Faction (AI Fix)

### PR DESCRIPTION
**Pull request by QuietlyWhisper**
## Zombie Faction (Zombie AI Fix)

Adds the zombie faction. All factions are hostile to zombies and vice-versa.
Zombified AI mobs will no longer attack other zombies!
TODO:
AI combat targets do not have their factions checked mid combat, which results in AI attacking zombified-on-death zombies until they are in crit. 

Zombified AI coming upon other zombies will not initiate combat, however!

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Whisper
- fix: Zombified AI will no longer attack other zombies.

### CODE REVIEW

**Comment by QuietlyWhisper** [comment]
It works, but is very dirty code that could be improved with better functions in factionsystem.cs
Also requires a check in NPCCombatOperator.cs, because AI will not update a target's faction until they are no longer a target

**Comment by QuietlyWhisper** [comment]
Not sure what else to do with this so I'll just leave it open for maintainers to look at ig

File: Content.Server/Zombies/ZombifyOnDeathSystem.cs [hunk 1]
```diff
@@ -92,6 +95,19 @@ public void ZombifyEntity(EntityUid target)
...
             var zombiecomp = AddComp<ZombieComponent>(target);
 
+            //Removes all other factions to prevent conflicting factions.
+            //This should ideally be coded to remove all factions rather than hardcoding them in... function does not exist.
+            //_faction.RemoveFaction(target);
+            _faction.RemoveFaction(target,"NanoTrasen", true);
+            _faction.RemoveFaction(target, "SimpleHostile", true);
+            _faction.RemoveFaction(target, "Syndicate", true);
+            _faction.RemoveFaction(target, "Xeno", true);
+            _faction.RemoveFaction(target, "Dragon", true);
```
Comment (AJCM-git) [inline Content.Server/Zombies/ZombifyOnDeathSystem.cs:105]: Just make a RemoveAllFactions method in faction system, it is a `Factions` hashset so it should be straightforward

File: Content.Server/Zombies/ZombifyOnDeathSystem.cs [hunk 2]
```diff
@@ -92,6 +95,19 @@ public void ZombifyEntity(EntityUid target)
...
+
+            //Adds the zombie faction, so ai zombies will no longer attack other zombies.
+            _faction.AddFaction(target, "Zombies");
+            _faction.MakeFriendly("Zombies", "Zombies");
```
Comment (AJCM-git) [inline Content.Server/Zombies/ZombifyOnDeathSystem.cs:109]: you can declare this in the prototype

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Server/Zombies/ZombifyOnDeathSystem.cs (hunk 1 above)
Comment (QuietlyWhisper) [reply Content.Server/Zombies/ZombifyOnDeathSystem.cs:105]: Dunno how.

File: Content.Server/NPC/Systems/FactionSystem.cs [hunk 3]
```diff
@@ -104,6 +104,18 @@ public void RemoveFaction(EntityUid uid, string faction, bool dirty = true)
...
+       /* /// <summary>
+        /// Removes all factions from the entity.
+        /// </summary>
+        public void RemoveFaction(EntityUid uid)
+        {
+            if (!TryComp<FactionComponent>(uid, out var component))
+                return;
+            if (!component.Factions.Remove())
+                return;
+        }*/
```
[the remark this answers is not in the document]
Comment (QuietlyWhisper) [reply Content.Server/NPC/Systems/FactionSystem.cs:117]: Dunno how.

**Pull request closed without merging**